Lib/ast.py

```diff
@@ -101,8 +102,20 @@ def _convert(node):
         return _convert_signed_num(node)
     return _convert(node_or_string)
 
+@lru_cache
+def _field_defaults(node_type):
```

Comment:
I don't understand this function. When we set the `node_type._field_defaults` we could already place `None` or `[]` in there instead of this intermediary "?" and "*".

Reply:
I dont think so, at least for lists. We are keeping `_field_defaults` as the same way we keep `_fields`, arrays of const char pointers. And then we construct a tuple from them and intern the values. I dont think it would be practical to do this with the original values.

Reply:
I don't fully understand the problem. What is wrong with having `_field_defaults` as a list of the defaults themselves? When constructing the `_field_defaults` tuple you can do this logic already and populate them with `None` or the empty list or whatever, no?

Reply:
Ah, that makes sense. Do you have an idea about what we should put if there is no field default, I think Ellipsis should work.

Reply:
You don't need that if you use a dictionary instead of a tuple